helm: get rid of storage group enablement based on the version

humblec opened the pull request:
- [x]  helm: get rid of storage group enablement based on the version
- [x]  helm: fuse_set_user_groups need not be part of the config
- [x]  deploy: remove beta1 mention from csidriver yaml
- [x] deploy : add missing fsgrouppolicy to th api and do generate-deploy

humblec committed:
helm: fuse_set_user_groups need not be part of the config

this setting in ceph.conf is no longer required and this commit
remove it from the chart.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
helm: get rid of storage group enablement based on the version
deploy: remove beta storage group mention from csidriver yaml

the kubernetes version based enablement of storage api group
enablement is no longer requried and its already on v1 for
supported kubernetes versions.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

Rakshith-R reviewed (commented):
remove comments from here too ? 
https://github.com/ceph/ceph-csi/blob/devel/deploy/cephfs/kubernetes/csidriver.yaml

humblec commented:
> remove comments from here too ? https://github.com/ceph/ceph-csi/blob/devel/deploy/cephfs/kubernetes/csidriver.yaml

sure. done.

Rakshith-R reviewed (changes requested):
here too
https://github.com/ceph/ceph-csi/blob/devel/api/deploy/kubernetes/rbd/csidriver.yaml#L2
and this may require autogeneration too

Rakshith-R reviewed (dismissed):
Thanks !

humblec committed:
deploy: add fsgrouppolicy to the driver yaml

this fsgrouppolicy setting was missing in api/deploy/* which
caused the yamlgen to not pickup this. this commit address the
same.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
@Rakshith-R one more commit added to have `fsgrouppolicy" in yamlgen source. ptal.

Madhu-1 reviewed (commented):
Please check if you can squash 2 and 3 commits into 1

humblec commented:
> Please check if you can squash 2 and 3 commits into 1

squashed 2 and 3 to one. @Madhu-1 ptal.. thanks

humblec commented:
@Mergifyio rebase

humblec commented:
The jenkins jobs are stuck. retriggering the tests

humblec commented:
/test ci/centos/k8s-e2e-external-storage/1.23

humblec commented:
https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/upgrade-tests-rbd/detail/upgrade-tests-rbd/6752/pipeline reserving the system fails in the CI. retesting.

humblec commented:
/test ci/centos/upgrade-tests-rbd

humblec commented:
/test ci/centos/mini-e2e-helm/k8s-1.23

humblec commented:
/test ci/centos/mini-e2e-helm/k8s-1.24

humblec commented:
/test ci/centos/mini-e2e-helm/k8s-1.25

humblec commented:
/test ci/centos/mini-e2e/k8s-1.25

humblec commented:
/test ci/centos/mini-e2e/k8s-1.24

humblec commented:
/test ci/centos/mini-e2e/k8s-1.23

humblec commented:
/test ci/centos/upgrade-tests-cephfs

humblec commented:
/test ci/centos/mini-e2e/k8s-1.23

nixpanic commented:
> https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/upgrade-tests-rbd/detail/upgrade-tests-rbd/6752/pipeline reserving the system fails in the CI. retesting.

When reserving times out, too many jobs are running. Most likely this PR is the cause, `ok-to-test` was added, and shortly after that the PR was rebased. CI jobs running for the pre-rebase version will not have finished when the `ok-to-test` label was added a 2nd time :disappointed:
